TimeStepping.pyx

```diff
@@ -127,6 +142,44 @@ cdef class TimeStepping:
 
         return
 
+    cpdef update_pressure_second(self, Grid.Grid Gr, PrognosticVariables.PrognosticVariables PV, DiagnosticVariables.DiagnosticVariables DV):
+        cdef:
+            Py_ssize_t i
+            Py_ssize_t press_shift = DV.get_varshift(Gr, 'density_perturbation_pressure')
+
+            Py_ssize_t whor_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve')
+            Py_ssize_t whor_rk0_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK0')
+            Py_ssize_t whor_rk1_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK1')
+
+            Py_ssize_t wpress_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient')
+            Py_ssize_t wpress_rk0_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK0')
+            Py_ssize_t wpress_rk1_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK1')
+
+
+        with nogil:
+            if self.rk_step == 0:
+                if self.t % self.dt_max == 0.0:
+                    for i in xrange(Gr.dims.npg):
+                        DV.values[whor_rk0_shift+i] = 0.0
+                        DV.values[wpress_rk0_shift+i] = 0.0
+                        DV.values[whor_rk1_shift+i] = 0.0
+                        DV.values[wpress_rk1_shift+i] = 0.0
+
```

Comment:
You can rewrite this more efficiently using array operations. E.g.:
` DV.values[whor_rk0_shift: whor_rk0_shift+Gr.dims.npg-1] = 0.0`

Also, **xrange()** is deprecated in python 3, so better to change it to **range()**